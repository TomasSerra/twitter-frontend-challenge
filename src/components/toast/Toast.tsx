import { useEffect, useState } from "react";
import { StyledToastContainer } from "./ToastContainer";
import { AlertIcon } from "../icon/Icon";
import { LightTheme } from "../../util/LightTheme";

export enum ToastType {
  ALERT = "ALERT",
  SUCCESS = "SUCCESS",
  WARNING = "WARNING",
  INFO = "INFO",
}

interface ToastProps {
  message: string;
  type: ToastType;
  show?: boolean;
  position?: "top-left" | "top-right" | "bottom-left" | "bottom-right";
  onClose?: () => void;
}

const Toast = ({
  message,
  type,
  show = true,
  position,
  onClose,
}: ToastProps) => {
  const [isShown, setIsShown] = useState<boolean>(show);

  useEffect(() => {
    if (!show) {
      setIsShown(false);
    }
  }, [show]);

  const handleClose = () => {
    setIsShown(false);
    if (onClose) onClose();
  };

  const iconMap = {
    [ToastType.ALERT]: <AlertIcon />,
    [ToastType.SUCCESS]: null,
    [ToastType.WARNING]: null,
    [ToastType.INFO]: null,
  };

  return isShown ? (
    <StyledToastContainer
      type={type}
      onClick={handleClose}
      position={position}
      theme={LightTheme}
    >
      {iconMap[type] || null}
      <p>{message}</p>
    </StyledToastContainer>
  ) : null;
};

export default Toast;
