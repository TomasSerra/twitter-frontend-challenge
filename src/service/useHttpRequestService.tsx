import { useState } from "react";
import axios from "axios";
import { S3Service } from "./S3Service";
import { useTranslation } from "react-i18next";
import { axiosInstance } from "./axiosInstance";
import type { PostData, SingInData, SingUpData } from "./index";

const url = process.env.REACT_APP_API_URL || "http://localhost:8081/api";

const useHttpRequestService = () => {
  const { t } = useTranslation();
  const [loading, setLoading] = useState<boolean>(false);
  const [error, setError] = useState<string | null>(null);

  const handleError = (error: any) => {
    if (axios.isAxiosError(error)) {
      if (error.response?.status === 404) {
        return [];
      }
      setError("Error fetching data");
      console.error("Error fetching:", error);
    }
  };

  const signUp = async (data: Partial<SingUpData>) => {
    setLoading(true);
    try {
      const res = await axiosInstance.post(`${url}/auth/signup`, data);
      if (res.status === 201) {
        localStorage.setItem("token", `Bearer ${res.data.token}`);
        return true;
      }
    } catch (e) {
      handleError(e);
      if (axios.isAxiosError(e)) {
        if (e.response?.status === 409) {
          throw new Error(t("error.signup.conflict"));
        }
        if (e.response?.status === 400) {
          throw new Error(t("error.signup.bad-request"));
        }
      }
      throw new Error(t("error.signup.error"));
    } finally {
      setLoading(false);
    }
  };

  const signIn = async (data: SingInData) => {
    setLoading(true);
    try {
      const res = await axiosInstance.post(`${url}/auth/login`, data);
      if (res.status === 200) {
        localStorage.setItem("token", `Bearer ${res.data.token}`);
        return true;
      }
    } catch (e) {
      handleError(e);
    } finally {
      setLoading(false);
    }
  };

  const createPost = async (data: PostData) => {
    setLoading(true);
    try {
      const uploadData = {
        content: data.content,
        images: data.images?.map((image) => image.name),
      };
      const res = await axiosInstance.post(`${url}/post`, uploadData);
      if (res.status === 201) {
        const { upload } = S3Service;
        for (const imageUrl of res.data.images) {
          const index: number = res.data.images.indexOf(imageUrl);
          console.log(imageUrl);
          await upload(data.images![index], imageUrl);
        }
        return res.data;
      }
    } catch (e) {
      handleError(e);
    } finally {
      setLoading(false);
    }
  };

  const getPaginatedPosts = async (
    limit: number,
    after: string,
    query: string
  ) => {
    try {
      const res = await axiosInstance.get(`${url}/post/${query}`, {
        params: {
          limit,
          after,
        },
      });
      if (res.status === 200) {
        return res.data;
      }
    } catch (e) {
      handleError(e);
    }
  };

  const getPosts = async (query: string) => {
    try {
      const res = await axiosInstance.get(`${url}/post/${query}`);
      if (res.status === 200) {
        return res.data;
      }
    } catch (e) {
      handleError(e);
    }
  };

  const getRecommendedUsers = async (limit: number, skip: number) => {
    try {
      const res = await axiosInstance.get(`${url}/user`, {
        params: {
          limit,
          skip,
        },
      });
      if (res.status === 200) {
        return res.data;
      }
    } catch (e) {
      handleError(e);
    }
  };

  const me = async () => {
    const res = await axiosInstance.get(`${url}/user/me`);
    if (res.status === 200) {
      return res.data.user;
    }
  };

  const getPostById = async (id: string) => {
    const res = await axiosInstance.get(`${url}/post/${id}`);
    if (res.status === 200) {
      return res.data;
    }
  };

  const createReaction = async (postId: string, reaction: string) => {
    const res = await axiosInstance.post(`${url}/reaction/${postId}`, {
      type: reaction,
    });
    if (res.status === 201) {
      return res.data;
    }
  };

  const deleteReaction = async (reactionId: string) => {
    const res = await axiosInstance.delete(`${url}/reaction/${reactionId}`);
    if (res.status === 200) {
      return res.data;
    }
  };

  const followUser = async (userId: string) => {
    const res = await axiosInstance.post(`${url}/follow/${userId}`);
    if (res.status === 201) {
      return res.data;
    }
  };

  const unfollowUser = async (userId: string) => {
    const res = await axiosInstance.delete(`${url}/follow/${userId}`);
    if (res.status === 200) {
      return res.data;
    }
  };

  const searchUsers = async (username: string, limit: number, skip: number) => {
    try {
      const cancelToken = axios.CancelToken.source();

      const response = await axiosInstance.get(`${url}/user/search`, {
        params: {
          username,
          limit,
          skip,
        },
        cancelToken: cancelToken.token,
      });

      if (response.status === 200) {
        return response.data;
      }
    } catch (error) {
      if (!axios.isCancel(error)) console.log(error);
      handleError(error);
    }
  };

  const getProfile = async (id: string) => {
    const res = await axiosInstance.get(`${url}/user/profile/${id}`);
    if (res.status === 200) {
      return res.data;
    }
  };

  const getPaginatedPostsFromProfile = async (
    limit: number,
    after: string,
    id: string
  ) => {
    try {
      const res = await axiosInstance.get(`${url}/post/by_user/${id}`, {
        params: {
          limit,
          after,
        },
      });

      if (res.status === 200) {
        return res.data;
      }
    } catch (e) {
      handleError(e);
    }
  };

  const getPostsFromProfile = async (id: string) => {
    try {
      const res = await axiosInstance.get(`${url}/post/by_user/${id}`);

      if (res.status === 200) {
        return res.data;
      }
    } catch (e) {
      handleError(e);
    }
  };

  const isLogged = async () => {
    try {
      const res = await axiosInstance.get(`${url}/user/me`);
      return res.status === 200;
    } catch (e) {
      return false;
    }
  };

  const getProfileView = async (id: string) => {
    const res = await axiosInstance.get(`${url}/user/${id}`);

    if (res.status === 200) {
      return res.data;
    }
  };

  const deleteProfile = async () => {
    const res = await axiosInstance.delete(`${url}/user/me`);

    if (res.status === 204) {
      localStorage.removeItem("token");
    }
  };

  const getChats = async () => {
    try {
      const res = await axiosInstance.get(`${url}/chat`);

      if (res.status === 200) {
        return res.data;
      }
    } catch (e) {
      handleError(e);
    }
  };

  const getMutualFollows = async () => {
    const res = await axiosInstance.get(`${url}/follow/mutual`);

    if (res.status === 200) {
      return res.data;
    }
  };

  const createChat = async (id: string) => {
    const res = await axiosInstance.post(`${url}/chat`, {
      users: [id],
    });

    if (res.status === 201) {
      return res.data;
    }
  };

  const getChat = async (id: string) => {
    const res = await axiosInstance.get(`${url}/chat/${id}`);

    if (res.status === 200) {
      return res.data;
    }
  };

  const deletePost = async (id: string) => {
    await axiosInstance.delete(`${url}/post/${id}`);
  };

  const getPaginatedCommentsByPostId = async (
    id: string,
    limit: number,
    after: string
  ) => {
    try {
      const res = await axiosInstance.get(`${url}/post/comment/by_post/${id}`, {
        params: {
          limit,
          after,
        },
      });
      if (res.status === 200) {
        return res.data;
      }
    } catch (e) {
      handleError(e);
    }
  };

  const getCommentsByPostId = async (id: string) => {
    try {
      const res = await axiosInstance.get(`${url}/post/comment/by_post/${id}`);
      if (res.status === 200) {
        return res.data;
      }
    } catch (e) {
      handleError(e);
    }
  };

  return {
    signUp,
    signIn,
    createPost,
    getPaginatedPosts,
    getPosts,
    getRecommendedUsers,
    me,
    getPostById,
    createReaction,
    deleteReaction,
    followUser,
    unfollowUser,
    searchUsers,
    getProfile,
    getPaginatedPostsFromProfile,
    getPostsFromProfile,
    isLogged,
    getProfileView,
    deleteProfile,
    getChats,
    getMutualFollows,
    createChat,
    getChat,
    deletePost,
    getPaginatedCommentsByPostId,
    getCommentsByPostId,
    loading,
    error,
  };
};

export default useHttpRequestService;
