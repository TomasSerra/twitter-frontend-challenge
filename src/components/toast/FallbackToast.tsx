import Toast, { ToastType } from "./Toast";

const FallbackToast = ({
  error,
  position,
}: {
  error: Error;
  position: "top-left" | "top-right" | "bottom-left" | "bottom-right";
}) => {
  return (
    <Toast type={ToastType.ALERT} message={error.message} position={position} />
  );
};

export default FallbackToast;
