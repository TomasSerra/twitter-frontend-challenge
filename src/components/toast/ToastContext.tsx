import {
  createContext,
  useContext,
  useState,
  useCallback,
  ReactNode,
} from "react";
import Toast, { ToastType } from "./Toast";

interface ToastMessage {
  id: number;
  type: ToastType;
  message: string;
  position?: "top-left" | "top-right" | "bottom-left" | "bottom-right";
}

interface ToastConfig {
  duration?: number;
  position?: "top-left" | "top-right" | "bottom-left" | "bottom-right";
}

interface ToastContextType {
  showToast: (
    type: ToastType,
    message: string,
    position?: ToastMessage["position"]
  ) => void;
}

const ToastContext = createContext<ToastContextType | undefined>(undefined);

export const ToastProvider = ({
  children,
  config = {},
}: {
  children: ReactNode;
  config?: ToastConfig;
}) => {
  const [toasts, setToasts] = useState<ToastMessage[]>([]);
  const { duration = 3000, position: defaultPosition = "top-right" } = config;

  const showToast = useCallback(
    (type: ToastType, message: string, position?: ToastMessage["position"]) => {
      const id = Date.now();
      const toastPosition = position || defaultPosition;

      setToasts((prev) => [
        ...prev,
        { id, type, message, position: toastPosition },
      ]);

      const timer = setTimeout(() => {
        setToasts((prev) => prev.filter((toast) => toast.id !== id));
      }, duration);

      return () => clearTimeout(timer);
    },
    [duration, defaultPosition]
  );

  const removeToast = (id: number) => {
    setToasts((prev) => prev.filter((toast) => toast.id !== id));
  };

  return (
    <ToastContext.Provider value={{ showToast }}>
      {children}
      {toasts.map((toast) => (
        <Toast
          key={toast.id}
          type={toast.type}
          message={toast.message}
          position={toast.position}
          onClose={() => removeToast(toast.id)}
        />
      ))}
    </ToastContext.Provider>
  );
};

export const useToast = () => {
  const context = useContext(ToastContext);
  if (!context) {
    throw new Error("useToast must be used inside a ToastProvider");
  }
  return context;
};
