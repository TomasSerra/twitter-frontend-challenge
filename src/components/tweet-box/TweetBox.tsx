import React, { useState } from "react";
import Button from "../button/Button";
import TweetInput from "../tweet-input/TweetInput";
import useHttpRequestService from "../../service/useHttpRequestService";
import { setLength, updateFeed } from "../../redux/user";
import ImageContainer from "../tweet/tweet-image/ImageContainer";
import { BackArrowIcon } from "../icon/Icon";
import ImageInput from "../common/ImageInput";
import { useTranslation } from "react-i18next";
import { ButtonColor, ButtonSize, ButtonType } from "../button/StyledButton";
import { StyledTweetBoxContainer } from "./TweetBoxContainer";
import { StyledContainer } from "../common/Container";
import { StyledButtonContainer } from "./ButtonContainer";
import { useDispatch, useSelector } from "react-redux";
import { useMe } from "../../hooks/useMe";
import { useToast } from "../toast/ToastContext";
import { ToastType } from "../toast/Toast";

interface TweetBoxProps {
  parentId?: string;
  close?: () => void;
  mobile?: boolean;
  borderless?: boolean;
}

const TweetBox = ({ parentId, close, mobile }: TweetBoxProps) => {
  const [content, setContent] = useState<string>("");
  const [images, setImages] = useState<File[]>([]);
  const [imagesPreview, setImagesPreview] = useState<string[]>([]);

  const { length, query } = useSelector(
    (state: { user: { length: number; query: string } }) => state.user
  );
  const { createPost, getPosts } = useHttpRequestService();
  const dispatch = useDispatch();
  const { t } = useTranslation();
  const { data: user } = useMe();
  const { showToast } = useToast();

  const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
    setContent(e.target.value);
  };

  const handleSubmit = async () => {
    try {
      await createPost({ content, images, parentId }).then(async (res) => {
        showToast(ToastType.SUCCESS, t("toast.post.create.success"));
        setContent("");
        setImages([]);
        setImagesPreview([]);
        dispatch(setLength(length + 1));
        const posts = await getPosts(query);
        dispatch(updateFeed(posts));
        close && close();
      });
    } catch (e) {
      showToast(ToastType.SUCCESS, t("toast.post.create.error"));
      console.error(e);
    }
  };

  const handleRemoveImage = (index: number) => {
    const newImages = images.filter((i, idx) => idx !== index);
    const newImagesPreview = newImages.map((i) => URL.createObjectURL(i));
    setImages(newImages);
    setImagesPreview(newImagesPreview);
  };

  const handleAddImage = (newImages: File[]) => {
    setImages(newImages);
    const newImagesPreview = newImages.map((i) => URL.createObjectURL(i));
    setImagesPreview(newImagesPreview);
  };

  return (
    <StyledTweetBoxContainer>
      {mobile && (
        <StyledContainer
          flexDirection={"row"}
          justifyContent={"space-between"}
          alignItems={"center"}
        >
          <BackArrowIcon onClick={close} />
          <Button
            buttonType={ButtonType.FULFILLED}
            size={ButtonSize.SMALL}
            buttonColor={ButtonColor.PRIMARY}
            onClick={handleSubmit}
            disabled={content.length === 0}
          >
            Tweet
          </Button>
        </StyledContainer>
      )}
      <StyledContainer style={{ width: "100%" }}>
        <TweetInput
          onChange={handleChange}
          maxLength={240}
          placeholder={t("placeholder.tweet")}
          value={content}
          src={user?.profilePicture}
        />
        <StyledContainer padding={"0 0 0 10%"}>
          <ImageContainer
            editable
            images={imagesPreview}
            removeFunction={handleRemoveImage}
          />
        </StyledContainer>
        <StyledButtonContainer>
          <ImageInput setImages={handleAddImage} parentId={parentId} />
          {!mobile && (
            <Button
              buttonType={ButtonType.FULFILLED}
              size={ButtonSize.SMALL}
              buttonColor={ButtonColor.PRIMARY}
              onClick={handleSubmit}
              disabled={
                content.length <= 0 ||
                content.length > 240 ||
                images.length > 4 ||
                images.length < 0
              }
            >
              Tweet
            </Button>
          )}
        </StyledButtonContainer>
      </StyledContainer>
    </StyledTweetBoxContainer>
  );
};

export default TweetBox;
