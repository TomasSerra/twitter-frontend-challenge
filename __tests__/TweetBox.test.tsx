import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import TweetBox from "../src/components/tweet-box/TweetBox";
import "@testing-library/jest-dom";
import { ThemeProvider } from "styled-components";
import useHttpRequestService from "../src/service/useHttpRequestService";
import { useToast } from "../src/components/toast/ToastContext";
import { ToastType } from "../src/components/toast/Toast";

jest.mock("../src/service/useHttpRequestService", () => {
  const createPost = jest.fn().mockResolvedValue({});
  const getPosts = jest.fn().mockResolvedValue([]);
  return {
    __esModule: true,
    default: jest.fn(() => ({
      createPost,
      getPosts,
    })),
  };
});

jest.mock("../src/redux/user", () => ({
  setLength: jest.fn(),
  updateFeed: jest.fn(),
}));

jest.mock("../src/hooks/useMe", () => ({
  useMe: () => ({ data: { profilePicture: "mockedProfilePic.jpg" } }),
}));

jest.mock("../src/components/tweet/tweet-image/ImageContainer", () => () => (
  <div data-testid="image-container"></div>
));

jest.mock("../src/components/icon/Icon", () => ({
  BackArrowIcon: () => <div data-testid="back-arrow-icon"></div>,
}));

jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

jest.mock("../src/components/toast/ToastContext", () => ({
  useToast: jest.fn(() => ({
    showToast: jest.fn(),
  })),
}));

jest.mock("react-redux", () => ({
  useDispatch: () => jest.fn(),
  useSelector: () => ({ length: 0, query: "" }),
}));

jest.mock("path/to/image.png", () => "test-image.png");

const mockTheme = {
  background: "#fff",
  colors: {
    main: "#4A99E9",
    light: "#A5CCF4",
    dark: "#428AD2",
    error: "#E03C39",
    white: "#FFFFFF",
    softWhite: "#f6f6f6",
    softPrimary: "#e5edf7",
    inactiveBackground: "#F0F3F4",
    containerLine: "#F0F3F4",
    hover: "#E7E7E8",
    outline: "#000000",
    text: "#566370",
    black: "#000000",
    errorContainer: "#E5397F",
    successContainer: "#4CAF50",
    warningContainer: "#FFC107",
    infoContainer: "#2196F3",
  },
  hover: {
    default: "#428AD2",
    main: "#428AD2",
    white: "#FFFFFF",
    follow: "#428AD2",
    error: "#FF0000",
    outlined: "#f3f3f3",
    disabled: "#A5CCF4",
  },
  text: {
    default: "#566370",
    title: "#000000",
    error: "#E03C39",
  },
  font: {
    default: "Manrope",
    title: "Inter",
  },
};

describe("TweetBox component", () => {
  let mockCreatePost: jest.Mock;
  let mockShowToast: jest.Mock;

  beforeEach(() => {
    mockCreatePost = jest.fn().mockResolvedValue({});
    mockShowToast = jest.fn();

    (useHttpRequestService as jest.Mock).mockReturnValue({
      createPost: mockCreatePost,
      getPosts: jest.fn().mockResolvedValue([]),
    });

    (useToast as jest.Mock).mockReturnValue({
      showToast: mockShowToast,
    });
  });

  test("TweetBox component renders correctly on PC", () => {
    const view = render(
      <ThemeProvider theme={mockTheme}>
        <TweetBox />
      </ThemeProvider>
    );
    expect(view).toMatchSnapshot();
    expect(
      screen.getByPlaceholderText("placeholder.tweet")
    ).toBeInTheDocument();

    expect(screen.getByTestId("tweet-button")).toBeInTheDocument();
  });

  test("TweetBox component renders correctly on mobile", () => {
    const view = render(
      <ThemeProvider theme={mockTheme}>
        <TweetBox mobile={true} />
      </ThemeProvider>
    );
    expect(view).toMatchSnapshot();
    expect(
      screen.getByPlaceholderText("placeholder.tweet")
    ).toBeInTheDocument();

    expect(screen.getByTestId("tweet-button-mobile")).toBeInTheDocument();
  });

  test("Tweet button is disabled when no content is entered", () => {
    render(
      <ThemeProvider theme={mockTheme}>
        <TweetBox mobile={false} />
      </ThemeProvider>
    );

    const tweetInput = screen.getByPlaceholderText(
      "placeholder.tweet"
    ) as HTMLInputElement;
    const tweetButton = screen.getByTestId("tweet-button");

    expect(tweetButton).toBeDisabled();

    fireEvent.change(tweetInput, { target: { value: "Nuevo tweet" } });

    expect(tweetButton).toBeEnabled();
  });

  test("content changes when typing in the input", () => {
    render(
      <ThemeProvider theme={mockTheme}>
        <TweetBox mobile={false} />
      </ThemeProvider>
    );

    const tweetInput = screen.getByPlaceholderText("placeholder.tweet");

    fireEvent.change(tweetInput, { target: { value: "Nuevo tweet" } });

    expect((tweetInput as HTMLInputElement).value).toBe("Nuevo tweet");
  });

  test("should call createPost when the tweet is submitted", async () => {
    render(
      <ThemeProvider theme={mockTheme}>
        <TweetBox />
      </ThemeProvider>
    );

    const input = screen.getByPlaceholderText("placeholder.tweet");

    fireEvent.change(input, { target: { value: "Test tweet" } });

    const button = screen.getByTestId("tweet-button");
    fireEvent.click(button);

    await waitFor(() => {
      expect(mockCreatePost).toHaveBeenCalledWith({
        content: "Test tweet",
        images: [],
        parentId: undefined,
      });
    });

    expect(mockShowToast).toHaveBeenCalledWith(
      ToastType.SUCCESS,
      "toast.post.create.success"
    );
  });

  test("should empty textarea when the tweet is submitted", async () => {
    render(
      <ThemeProvider theme={mockTheme}>
        <TweetBox />
      </ThemeProvider>
    );

    const input = screen.getByPlaceholderText("placeholder.tweet");

    fireEvent.change(input, { target: { value: "Test tweet" } });

    const button = screen.getByTestId("tweet-button");
    fireEvent.click(button);

    await waitFor(() => {
      expect((input as HTMLInputElement).value).toBe("");
    });
  });

  test("should throw error in toast when the tweet is not submitted", async () => {
    (useHttpRequestService as jest.Mock).mockReturnValue({
      createPost: jest.fn().mockRejectedValue({}),
      getPosts: jest.fn(),
    });

    render(
      <ThemeProvider theme={mockTheme}>
        <TweetBox />
      </ThemeProvider>
    );

    const input = screen.getByPlaceholderText("placeholder.tweet");

    fireEvent.change(input, { target: { value: "Test tweet" } });

    const button = screen.getByTestId("tweet-button");
    fireEvent.click(button);

    await waitFor(() => {
      expect(mockShowToast).toHaveBeenCalledWith(
        ToastType.ALERT,
        "toast.post.create.error"
      );
    });
  });

  test("Should allow adding an image", async () => {
    render(
      <ThemeProvider theme={mockTheme}>
        <TweetBox />
      </ThemeProvider>
    );

    const file = new File(["image-content"], "test-tweet-image.png", {
      type: "image/png",
    });
    const fileInput = screen.getByTestId("image-input");

    fireEvent.change(fileInput, { target: { files: [file] } });

    await waitFor(() => {
      expect(screen.getByTestId("tweet-image")).toBeInTheDocument();
    });
  });
});
