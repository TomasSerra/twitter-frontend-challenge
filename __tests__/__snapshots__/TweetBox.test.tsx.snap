// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TweetBox component TweetBox component renders correctly on PC 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-ispOId eotdWM"
      >
        <div
          class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
          style="width: 100%;"
        >
          <div
            class="sc-ewnqHT cJfwNi"
          >
            <div
              class="sc-gTRrQi bFBxc"
            >
              <img
                alt="Icon"
                src="mockedProfilePic.jpg"
              />
            </div>
            <textarea
              class="sc-fFGjHI jcvhjS"
              maxlength="240"
              placeholder="placeholder.tweet"
            />
          </div>
          <div
            class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
            style="padding: 0px 0px 0px 10%;"
          >
            <div
              data-testid="image-container"
            />
          </div>
          <div
            class="sc-kcuKUB ljiXJx"
          >
            <div
              class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
              cursor="pointer"
              style="justify-content: center; padding: 5px 10px; cursor: pointer;"
              width="fit-content"
            >
              <label
                for="image-upload-"
                style="cursor: pointer;"
              >
                <img
                  alt="Upload Images"
                  src="test-image.png"
                />
              </label>
              <input
                accept="image/*"
                id="image-upload-"
                multiple=""
                style="display: none;"
                type="file"
              />
            </div>
            <button
              class="sc-bgqQcB jXgpsi"
              data-testid="tweet-button"
              disabled=""
            >
              <span>
                Tweet
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-ispOId eotdWM"
    >
      <div
        class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
        style="width: 100%;"
      >
        <div
          class="sc-ewnqHT cJfwNi"
        >
          <div
            class="sc-gTRrQi bFBxc"
          >
            <img
              alt="Icon"
              src="mockedProfilePic.jpg"
            />
          </div>
          <textarea
            class="sc-fFGjHI jcvhjS"
            maxlength="240"
            placeholder="placeholder.tweet"
          />
        </div>
        <div
          class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
          style="padding: 0px 0px 0px 10%;"
        >
          <div
            data-testid="image-container"
          />
        </div>
        <div
          class="sc-kcuKUB ljiXJx"
        >
          <div
            class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
            cursor="pointer"
            style="justify-content: center; padding: 5px 10px; cursor: pointer;"
            width="fit-content"
          >
            <label
              for="image-upload-"
              style="cursor: pointer;"
            >
              <img
                alt="Upload Images"
                src="test-image.png"
              />
            </label>
            <input
              accept="image/*"
              id="image-upload-"
              multiple=""
              style="display: none;"
              type="file"
            />
          </div>
          <button
            class="sc-bgqQcB jXgpsi"
            data-testid="tweet-button"
            disabled=""
          >
            <span>
              Tweet
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TweetBox component TweetBox component renders correctly on mobile 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-ispOId eotdWM"
      >
        <div
          class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
          style="flex-direction: row; justify-content: space-between; align-items: center;"
        >
          <div
            data-testid="back-arrow-icon"
          />
          <button
            class="sc-bgqQcB jXgpsi"
            data-testid="tweet-button-mobile"
            disabled=""
          >
            <span>
              Tweet
            </span>
          </button>
        </div>
        <div
          class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
          style="width: 100%;"
        >
          <div
            class="sc-ewnqHT cJfwNi"
          >
            <div
              class="sc-gTRrQi bFBxc"
            >
              <img
                alt="Icon"
                src="mockedProfilePic.jpg"
              />
            </div>
            <textarea
              class="sc-fFGjHI jcvhjS"
              maxlength="240"
              placeholder="placeholder.tweet"
            />
          </div>
          <div
            class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
            style="padding: 0px 0px 0px 10%;"
          >
            <div
              data-testid="image-container"
            />
          </div>
          <div
            class="sc-kcuKUB ljiXJx"
          >
            <div
              class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
              cursor="pointer"
              style="justify-content: center; padding: 5px 10px; cursor: pointer;"
              width="fit-content"
            >
              <label
                for="image-upload-"
                style="cursor: pointer;"
              >
                <img
                  alt="Upload Images"
                  src="test-image.png"
                />
              </label>
              <input
                accept="image/*"
                id="image-upload-"
                multiple=""
                style="display: none;"
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-ispOId eotdWM"
    >
      <div
        class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
        style="flex-direction: row; justify-content: space-between; align-items: center;"
      >
        <div
          data-testid="back-arrow-icon"
        />
        <button
          class="sc-bgqQcB jXgpsi"
          data-testid="tweet-button-mobile"
          disabled=""
        >
          <span>
            Tweet
          </span>
        </button>
      </div>
      <div
        class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
        style="width: 100%;"
      >
        <div
          class="sc-ewnqHT cJfwNi"
        >
          <div
            class="sc-gTRrQi bFBxc"
          >
            <img
              alt="Icon"
              src="mockedProfilePic.jpg"
            />
          </div>
          <textarea
            class="sc-fFGjHI jcvhjS"
            maxlength="240"
            placeholder="placeholder.tweet"
          />
        </div>
        <div
          class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
          style="padding: 0px 0px 0px 10%;"
        >
          <div
            data-testid="image-container"
          />
        </div>
        <div
          class="sc-kcuKUB ljiXJx"
        >
          <div
            class="sc-dicizt sc-iVCKna IPeKc dAgTYq"
            cursor="pointer"
            style="justify-content: center; padding: 5px 10px; cursor: pointer;"
            width="fit-content"
          >
            <label
              for="image-upload-"
              style="cursor: pointer;"
            >
              <img
                alt="Upload Images"
                src="test-image.png"
              />
            </label>
            <input
              accept="image/*"
              id="image-upload-"
              multiple=""
              style="display: none;"
              type="file"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
